javatests/arcs/core/data/RecipeTest.kt

```diff
@@ -289,6 +290,18 @@ class RecipeTest {
                         )
                     )
                 ),
+                handles = listOf(
+                    Plan.Handle(
+                        storageKey = StorageKeyParser.parse(peopleStorageKey),
+                        type = personCollectionType,
```

Comment:
Wouldn't a better test be to have the HandleConnection type be a subtype of the Handle type? The original problem was you could have two connections attached to the same handle, type T and U, and the handle would get assigned the first one processed (either T or U).   Like if you inserted an extra connection that had a different type, but the same storageKey it would be a more effective test I think.

Reply:
Yes, it would be better, but it would also complicate the this test a fair bit. Instead of complicating this E2E test I've added a dedicated test for the Recipe.Handle -> Plan.Handle does tests the conversion and shows it happens without the consideration of what are the types of the handle connections, which is what we care about I think.